From abeed615777690ba748f64bdb72a35fe5311259b Mon Sep 17 00:00:00 2001
From: Rodrigo Arias <rodrigo.arias@bsc.es>
Date: Thu, 25 Aug 2022 16:36:38 +0200
Subject: [PATCH] Emit a fill event at shutdown

---
 src/instrument/ovni/InstrumentThreadManagement.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/instrument/ovni/InstrumentThreadManagement.cpp b/src/instrument/ovni/InstrumentThreadManagement.cpp
index c79a9750..4872b2d9 100644
--- a/src/instrument/ovni/InstrumentThreadManagement.cpp
+++ b/src/instrument/ovni/InstrumentThreadManagement.cpp
@@ -92,6 +92,12 @@ void Instrument::threadHasResumed(__attribute__((unused)) external_thread_id_t t
 
 void Instrument::threadWillShutdown()
 {
+	ThreadLocalData &tld = getThreadLocalData();
+	if (tld._hungry) {
+		tld._hungry = false;
+		Ovni::schedFill();
+	}
+
 	Ovni::threadEnd();
 }
 
-- 
2.36.1

